### PR TITLE
Make example trace output optional via `--save-runs`

### LINKED ISSUE #2039: Performance regression outputting an example trace
We added default output of an example trace in #1838. It is implemented by the call to `trex.sat()` below.

https://github.com/informalsystems/apalache/blob/eb92cc88883ca2134c06ba6ab644b8d60585d4bf/tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/SeqModelChecker.scala#L95

This may cause a significant performance regression, as witnessed by #2019, where producing the example trace takes one hour.

In addition, when Apalache is run with `--discard-disabled`, it might overshoot the actually feasible trace length. This essentially causes an unsat query in the `trex.sat()` call that takes substantial time. Also, the reason for the delayed behavior is not obvious from the tool output:

```
$ apalache-mc check --features=rows --length=9 --inv=Mutex --discard-disabled=false --no-deadlock --tuning-options='search.transitionFilter=(0->.*|[1-6]->.*|7->3|[8-9]->.*|[1-9][0-9]+->.*)' test/tla/MC_LamportMutexTyped.tla
...
13:12:43.344 [main] DEBUG a.f.a.t.b.t.TransitionExecutorImpl - Step #9, transition #5
13:12:43.345 [main] DEBUG a.f.a.t.b.t.TransitionExecutorImpl - Translating to SMT...
13:12:43.442 [main] INFO  a.f.a.t.b.t.TransitionExecutorImpl - Step 9: picking a transition out of 6 transition(s)
<<note the delay here>>
13:16:34.672 [main] INFO  a.f.a.t.b.p.BoundedCheckerPassImpl - The outcome is: NoError
13:16:34.672 [main] DEBUG a.f.a.i.p.PassChainExecutor - PASS #13: BoundedChecker [OK]
```

### PR DESCRIPTION
**Pull request by thpani**
We introduced example trace output in #1838. As described in #2039 and #2019, this creates a unexpected performance bottleneck if the SMT call for constructing the trace takes a long time.

Oddly enough, writing the trace was made optional for `simulate` via the `--save-runs` flag   (defaulting to `false`) in #1838, but always enabled for `check`.

With this PR, we lift `--save-runs` to `check`, making example trace output optional also for `check`.

We also make some UX improvements that
* fix the case where the last symbolic run of `simulate` was output twice,
* add log output around the example trace output, and
* adjust example output log levels from `ERROR` to `INFO`.

- [ ] ~Tests added for any new code~
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] ~Documentation added for any new functionality~
- [x] [Entries added to `./unreleased/`][changelog format] for any new functionality

[changelog format]: https://github.com/informalsystems/apalache/blob/main/CONTRIBUTING.md#how-to-record-a-change

Fixes #2039

### CODE REVIEW

File: mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala [hunk 1]
```diff
@@ -56,6 +56,10 @@ class CheckCmd(name: String = "check", description: String = "Check a TLA+ speci
     opt[String](name = "view", description = "the state view to use with --max-error=n, default: transition index",
         default = "")
 
+  var saveRuns: Boolean =
+    opt[Boolean](name = "save-runs", description = "save an example trace for each symbolic run, default: false",
+        default = false)
```
Comment (Kukovec) [inline mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala:61]: I feel like the name "save-runs" isn't really correct, at least in the non-simulation case. Would it make sense to rename the flag in both cases to e.g. `enable-traces` or `tracing`? Also, the description here is not really suitable for `check` IMO ("each" is confusing here).

Comment (thpani) [reply mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala:61]: > I feel like the name "save-runs" isn't really correct, at least in the non-simulation case

Why do you think that's the case? It's doing the same thing, right?

> Also, the description here is not really suitable for check IMO ("each" is confusing here).

Do you think "each" suggests that there are multiple?

I think Igor's idea was that we do not use the word "trace" with users.

Comment (thpani) [reply mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala:61]: How about `--dump-traces`?

**Comment by thpani** [comment]
Had to do some wiring in in [f3a5b2b](https://github.com/informalsystems/apalache/pull/2047/commits/f3a5b2b2d1e08d012f95182d64b781c8ec4c05f1).
I find it weird that this is wired in via tuning options, but we can fix that in a follow-up PR.

File: mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala (hunk 1 above)
Comment (Kukovec) [reply mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala:61]: Maybe `output-`, rather than `dump-`.

**Review by Kukovec — APPROVED** [review]

File: mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala (hunk 1 above)
Comment (thpani) [reply mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala]: Renamed in [63494ad](https://github.com/informalsystems/apalache/pull/2047/commits/63494ad4fffe6cba3443127ed9ea3566338d4c63)

**Pull request merged**